### ISSUE TITLE
Script exiting with refrence error

### ISSUE THREAD

**Issue by ostersc** [tsightler/ring-mqtt#3]
The script runs for me, successfully listing out devices with the printouts of "HASS config topic:" but then outputs this, and exits.

2019-03-17T14:27:17.752Z ring-alarm-mqtt Exit code: ReferenceError: error is not defined
2019-03-17T14:27:18.388Z ring-alarm-mqtt ring/alarm/dphgg4-306bo-0/alarm_control_panel/2179a162-f20a-42e2-a0aa-81c8c9197827/state disarmed
2019-03-17T14:27:18.392Z ring-alarm-mqtt ring/alarm/dphgg4-306bo-0/binary_sensor/f1a321ba-98f1-4440-9eb2-1a92d794c0ba/state OFF
2019-03-17T14:27:18.393Z ring-alarm-mqtt ring/alarm/dphgg4-306bo-0/binary_sensor/d2d0ad2a-1671-4588-9159-8aca6fcecdc1/state ON
2019-03-17T14:27:18.395Z ring-alarm-mqtt ring/alarm/dphgg4-306bo-0/binary_sensor/8ead3ef2-996a-4ac9-a9c2-f4b59abded8e/state OFF
2019-03-17T14:27:18.396Z ring-alarm-mqtt ring/alarm/dphgg4-306bo-0/binary_sensor/f4cb288b-4b1e-4437-abf9-c0c6018592f1/state OFF
2019-03-17T14:27:18.398Z ring-alarm-mqtt ring/alarm/dphgg4-306bo-0/binary_sensor/fa2f9f6d-0c84-4590-8851-aada9bca6f9c/state ON
2019-03-17T14:27:18.399Z ring-alarm-mqtt ring/alarm/dphgg4-306bo-0/binary_sensor/6324671c-c7fd-42d9-94c4-088df407e7f3/state OFF
2019-03-17T14:27:18.403Z ring-alarm-mqtt ring/alarm/dphgg4-306bo-0/binary_sensor/4ab90997-baa3-4d68-9d23-9c6d977d1e03/state OFF
2019-03-17T14:27:18.756Z ring-alarm-mqtt Exit code: 0

**Comment by tsightler** [comment]
All of this is basically correct, it's the state for each sensor and that output is expected.  What's not expected is exiting unless you canceled the script.  Can you run the script with full debug and see if there are any obvious errors?  Also, what version of nodeJS are you running on?

**Comment by ostersc** [comment]
It looks like there's probably a syntax/scoping issue with the error handling code when the mqtt connection fails.  I found an authentication error in my mqtt broker's logs, and noticed I had my password wrong.  So that was the root cause of the error, and when I corrected that it started working.  So I'm good now, but there's probably an issue in the code that is preventing the script from properly outputting the error as intended.  Thanks for the great project!

**Comment by tsightler** [comment]
Ok, I'll test this scenario more stringently and dig up the error. I definitely saw that the error was not getting reported but obviously that didn't cause the exit.  I've actually tested failure scenarios a good bit but maybe missed the MQTT fails to connect on startup case.

Thanks for the report!

**Comment by tsightler** [comment]
I found the error, it was silly!  Just missing the "error" in the function parameter.  I had tested with the MQTT broker not running/reachable, but I had not tested a scenario where the broker was connected, but not authorized to send data.

Once I added the error parameter to the function it now generates a nice "Not authorized" message (at least if you have debug enabled) and, instead of exiting, it just keeps trying to connect every second like it's supposed to.  I'll push an update to the dev branch to fix this, so that when I push that to the master branch later next week it gets fixed.  Thanks again for reporting it!
